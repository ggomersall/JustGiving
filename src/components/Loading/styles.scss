.loading__wrapper {
  margin: 100px auto;
  display: block;
  width: 50%;
  text-align: center;

  .loading__text {
    margin-top: 20px;
  }
}