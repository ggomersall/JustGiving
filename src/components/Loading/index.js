import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@emotion/core';
import {ClipLoader} from 'react-spinners';

import './styles.scss';


const Loading = ({loading}) => {

  const override = css `
    display: block;
    margin: 0 auto;
    border-color: red;
  `;

  return (
    <div className="loading__wrapper">
      <ClipLoader 
        loading={loading} 
        css={override}
        sizeUnit={'px'}
        size={70}
        color={'#882c84'}/>
      <p className="loading__text">Loading</p>
    </div>
  );
};

Loading.propTypes = {
  loading: PropTypes.bool
};

export default Loading;