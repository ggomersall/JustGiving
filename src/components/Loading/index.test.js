import React from 'react';
import { shallow } from 'enzyme';
import Loading from './index.js';

describe('Testing Loading Component', () => {

  let wrapper;

  it('renders correctly', () => {
    shallow(<Loading />);
  });

  test('it should render loading props correctly', () => {
    const loading = true;
    wrapper = shallow(<Loading loading={loading} />);
  });

  test('Snapshot', () => expect(<Loading />).toMatchSnapshot());

});

