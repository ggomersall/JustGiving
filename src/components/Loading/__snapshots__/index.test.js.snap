// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Loading Component Snapshot 1`] = `<Loading />`;
